File: non-native/service_buddy/lib/models/maintenance_event.dart
import 'package:uuid/uuid.dart';

class MaintenanceEvent {
  String id;
  String vehicleIdentifier;
  String title;
  String? description;
  String category;
  DateTime dueDate;
  String status;
  double price;

  MaintenanceEvent({
    String? id,
    required this.vehicleIdentifier,
    required this.title,
    this.description,
    required this.category,
    required this.dueDate,
    required this.status,
    required this.price,
  }) : id = id ?? const Uuid().v4();
}

File: native/ServiceBuddy/app/src/main/java/com/example/servicebuddy/model/MaintenanceEvent.kt
package com.example.servicebuddy.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date
import java.util.UUID

@Entity(tableName = "events")
data class MaintenanceEvent(
    @PrimaryKey
    val id: String = UUID.randomUUID().toString(),
    var vehicleIdentifier: String,
    var title: String,
    var description: String?,
    var category: String,
    var dueDate: Date,
    var status: String,
    var price: Double,

    var isDirty: Boolean = false,         
    var isDeletedLocally: Boolean = false 
)
